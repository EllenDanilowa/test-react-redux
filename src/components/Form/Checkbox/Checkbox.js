import React from 'react';
import PropTypes from 'prop-types';
import {
  CheckboxWrapper,
  StyledCheckbox,
  Icon,
  Title
} from './Checkbox.styled';
import {
  FieldWrapper,
  Label,
  HiddenInput
} from '../Form.styled';

const Checkbox = ({refFunc, title, name, checked}) => (
  <FieldWrapper>
    <Label htmlFor={name}>
      <CheckboxWrapper>
        <HiddenInput ref={refFunc}
                     name={name}
                     id={name}
                     type="checkbox"/>
        <StyledCheckbox checked={checked}>
          <Icon viewBox="0 0 24 24" >
            <polyline points="20 6 9 17 4 12"/>
          </Icon>
        </StyledCheckbox>
        <Title>{title}</Title>
      </CheckboxWrapper>
    </Label>
  </FieldWrapper>
);

Checkbox.propTypes = {
  refFunc: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  checked: PropTypes.bool
};

export default Checkbox;
