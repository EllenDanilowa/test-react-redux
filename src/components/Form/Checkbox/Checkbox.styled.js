import styled from 'styled-components';
import colors from '../../../styles/colors';

export const CheckboxWrapper = styled.div`
  display: inline-block;
  position: relative;
  vertical-align: middle;
`;

export const Icon = styled.svg`
  fill: none;
  stroke: ${colors.main_background};
  stroke-width: 2px;
`;

export const HiddenCheckbox = styled.input.attrs({type: 'checkbox'})`
  border: 0;
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
`;

export const Title = styled.span`
  margin-left: 9px;
`;

export const StyledCheckbox = styled.div`
  display: inline-block;
  border: 1px solid ${colors.main_background};
  border-radius: 3px;
  height: 21px;
  width: 21px;

  ${Icon} {
    visibility: ${(props) => (props.checked ? 'visible' : 'hidden')}
  }
`;
