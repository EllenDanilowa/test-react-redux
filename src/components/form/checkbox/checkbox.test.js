import React from 'react';
import renderer from 'react-test-renderer';
import Checkbox from './checkbox';

describe('Checkbox', () => {
  let element;
  let title;
  let name;
  let ref;
  let checked;

  beforeEach(() => {
    title = 'Title';
    name = 'name';
    ref = () => {};
    checked = true;
  });

  it('renders component with defined props', () => {
    element = renderer.create(<Checkbox refFunc={ref} name={name} title={title} checked={checked}/>).toJSON();

    expect(element).toMatchSnapshot();
  });
});
