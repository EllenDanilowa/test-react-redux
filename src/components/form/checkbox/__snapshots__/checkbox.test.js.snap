// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders component with defined props 1`] = `
<div
  className="sc-AxgMl hnlgA-D"
>
  <label
    className="sc-AxhUy lTvU"
    htmlFor="name"
  >
    <div
      className="sc-AxjAm gZeZUl"
    >
      <input
        className="sc-AxheI icGmwi"
        id="name"
        name="name"
        type="checkbox"
      />
      <div
        checked={true}
        className="sc-AxhCb dJfVvl"
      >
        <svg
          className="sc-AxirZ cZmLyS"
          viewBox="0 0 24 24"
        >
          <polyline
            points="20 6 9 17 4 12"
          />
        </svg>
      </div>
      <span
        className="sc-AxiKw iNXYxW"
      >
        Title
      </span>
    </div>
  </label>
</div>
`;
